encrypt/ecies/ecies.go

```diff
@@ -0,0 +1,114 @@
+// Package ecies implements the Elliptic Curve Integrated Encryption Scheme (ECIES).
+package ecies
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha256"
+	"errors"
+	"hash"
+
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/util/random"
+	"golang.org/x/crypto/hkdf"
+)
+
+// Encrypt first computes a shared DH key using the given public key, then
+// HKDF-derives a symmetric key (and nonce) from that, and finally uses these
+// values to encrypt the given message via AES-GCM. If the hash input parameter
+// is nil then SHA256 is used as a default. Encrypt returns the ephemeral
+// elliptic curve point of the DH key exchange and the ciphertext or an error.
```

Comment:
The output is a point and a ciphertext, or an error. So why is the signature not (eph kyber,Point, ctx []byte, err error)?

I think I can see why: because the ephemeral point will be sent across the wire anyway, and we've already marshalled it into Rb, so why throw that work away and make the caller repeat it?

But then why not return one sincel []byte, which is ready to be sent across the wire? Why would the caller be interested in having the Rb separate from the ciphertext?

This is more curiosity on my part, I don't think it needs to be changed from how it is, really.

If this was in the critical path for performance, it would need to be reworked for the API to allow reuse of existing buffers in order to create less garbage. If you think that's important ask and I can point out how.

Reply:
I also asked my self this question if we should just return a pure slice of bytes directly, in the spirit of other implementation that we have in kyber (such as tlbs and bls).

I can't find an appropriate use of this Rb alone. It's maybe important to note that here, this ECIES scheme only "authenticates" the receiver, the sender does not use any private information. Doing "mutual authentication" in the future is possible by adding a or multiple signatures from the sender. One signature could be over this Rb, but it's only one choice amongst other possibilities that seem equivalently secure. 
So for the sake of clarity, I'm also supporting the change to just return one slice of byte containing everything.

Reply:
Thanks for all your comments. I agree that there seems to be no obvious advantage to keep the ephemeral point and ciphertext separate, I will change that so that the function just returns a single byte slice.

Reply:
So I understand you are going to change this still, so I'm moving back to WIP.